src/codemagic/tools/app_store_connect/actions/fetch_signing_files_action.py

```diff
@@ -0,0 +1,237 @@
+from codemagic.apple.resources import DeviceStatus
+from codemagic.apple.resources import Profile
+from codemagic.apple.resources import ProfileState
+from codemagic.apple.resources import ProfileType
+from codemagic.apple.resources import SigningCertificate
+from codemagic.cli import Colors
+from codemagic.models import PrivateKey
+
+from ..abstract_base_action import AbstractBaseAction
+from ..arguments import BundleIdArgument
+from ..arguments import CertificateArgument
+from ..arguments import CommonArgument
+from ..arguments import ProfileArgument
+from ..arguments import Types
+from ..errors import AppStoreConnectError
+
+
+class FetchSigningFilesAction(AbstractBaseAction, metaclass=ABCMeta):
+    @cli.action(
+        "fetch-signing-files",
+        BundleIdArgument.BUNDLE_ID_IDENTIFIER,
+        BundleIdArgument.PLATFORM,
+        CertificateArgument.PRIVATE_KEY,
+        CertificateArgument.PRIVATE_KEY_PASSWORD,
+        CertificateArgument.P12_CONTAINER_PASSWORD,
+        ProfileArgument.PROFILE_TYPE,
+        BundleIdArgument.IDENTIFIER_STRICT_MATCH,
+        CommonArgument.CREATE_RESOURCE,
+    )
+    def fetch_signing_files(
+        self,
+        bundle_id_identifier: str,
+        certificate_key: Optional[Union[PrivateKey, Types.CertificateKeyArgument]] = None,
+        certificate_key_password: Optional[Types.CertificateKeyPasswordArgument] = None,
+        p12_container_password: str = "",
+        platform: BundleIdPlatform = BundleIdPlatform.IOS,
+        profile_type: ProfileType = ProfileType.IOS_APP_DEVELOPMENT,
+        bundle_id_identifier_strict_match: bool = False,
+        create_resource: bool = False,
+    ) -> Tuple[List[Profile], List[SigningCertificate]]:
+        """
+        Fetch provisioning profiles and code signing certificates
+        for Bundle ID with given identifier
+        """
+
+        private_key = self._get_certificate_key(certificate_key, certificate_key_password)
+        if private_key is None:
+            raise AppStoreConnectError(f"Cannot save {SigningCertificate.s} without certificate private key")
+
+        bundle_ids = self._get_or_create_bundle_ids(
+            bundle_id_identifier,
+            platform,
+            create_resource,
+            bundle_id_identifier_strict_match,
+        )
+        self.echo("")
+
+        certificates = self._get_or_create_certificates(
+            profile_type,
+            certificate_key,
+            certificate_key_password,
+            create_resource,
+        )
+        self.echo("")
+
+        profiles = self._get_or_create_profiles(
+            bundle_ids,
+            certificates,
+            profile_type,
+            create_resource,
+            platform,
+        )
+        self.echo("")
+
+        self._save_certificates(certificates, private_key, p12_container_password)
+        self._save_profiles(profiles)
+        return profiles, certificates
+
+    def _get_or_create_bundle_ids(
```

Comment:
Used to be defined in https://github.com/codemagic-ci-cd/cli-tools/blob/3670fa624997f17cb437bc88198f7bdee3594f35/src/codemagic/tools/app_store_connect.py#L885-L905